wgpu-core/src/device/mod.rs

```diff
@@ -1440,21 +1440,26 @@ impl<B: GfxBackend> Device<B> {
                                     view_samples: view.samples as u32,
                                 });
                             }
-                            match (sample_type, format_info.sample_type) {
-                                (Tst::Uint, Tst::Uint) |
-                                (Tst::Sint, Tst::Sint) |
-                                (Tst::Depth, Tst::Depth) |
-                                // if we expect non-fiterable, accept anything float
-                                (Tst::Float { filterable: false }, Tst::Float { .. }) |
-                                // if we expect fiterable, require it
-                                (Tst::Float { filterable: true }, Tst::Float { filterable: true }) |
+                            match (sample_type, format_info.sample_type, view.format_features.filterable ) {
```

Comment:
I think this part can be refactored a bit, but I don't want to hold it. Your fix looks good!